ParcelDistributionCenter/ParcelDistributionCenter.Web/Controllers/PackagesController.cs

```diff
@@ -37,16 +40,24 @@ public ActionResult AddPackage()
         [ValidateAntiForgeryToken]
         public ActionResult AddPackage(PackageViewModel packageViewModel)
         {
-            if (ModelState.IsValid)
+            try
             {
-                packageViewModel = _addNewPackageService.AddNewPackage(packageViewModel);
-                //TODO: TempData do przeniesienia do widoku
-                TempData["Message"] = "Package successfully added!";
-                TempData["MessageClass"] = "alert-success";
+                if (ModelState.IsValid)
+                {
+                    packageViewModel = _addNewPackageService.AddNewPackage(packageViewModel);
+                    //TODO: TempData do przeniesienia do widoku
+                    TempData["Message"] = "Package successfully added!";
+                    TempData["MessageClass"] = "alert-success";
 
-                return RedirectToAction(nameof(DisplaySinglePackage), packageViewModel);
+                    return RedirectToAction(nameof(DisplaySinglePackage), packageViewModel);
+                }
+                return View(packageViewModel);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex.Message);
+                return StatusCode(500);
```

Comment:
Może zamiast rzucać 500, wyświetlić komunikat:
TempData["Message"] = "Sorry. Something went wrong. We will contact you to resolve issue.";
TempData["MessageClass"] = "alert-danger";
return RedirectToAction(nameof(DisplaySinglePackage), packageViewModel);

Reply:
Spoko, zmienione :D